public/app/percona/dbaas/components/DBCluster/DBCluster.utils.test.tsx

```diff
@@ -98,4 +105,76 @@ describe('DBCluster.utils::', () => {
     expect(isOptionEmpty({ label: 'test label' })).toBeTruthy();
     expect(isOptionEmpty({ value: 'test value' })).toBeFalsy();
   });
+  it('calculates resources difference correctly', () => {
+    const resourcesA = {
+      value: 10,
+      original: 10,
+      units: ResourcesUnits.BYTES,
+    };
+    const resourcesB = {
+      value: 20,
+      original: 20,
+      units: ResourcesUnits.BYTES,
+    };
+    const resourcesC = {
+      value: 20,
+      original: 20,
+      units: ResourcesUnits.GB,
+    };
+    const expectedResourcesA = {
+      expected: {
+        cpu: resourcesA,
+        memory: resourcesA,
+        disk: resourcesA,
+      },
+    };
+    const expectedResourcesB = {
+      expected: {
+        cpu: resourcesB,
+        memory: resourcesB,
+        disk: resourcesB,
+      },
+    };
+    const resultA = {
+      value: 0,
+      original: 0,
+      units: ResourcesUnits.BYTES,
+    };
+    const resultB = {
+      value: -10,
+      original: -10,
+      units: ResourcesUnits.BYTES,
+    };
+    const resultC = {
+      value: 10,
+      original: 10,
+      units: ResourcesUnits.BYTES,
+    };
+
+    expect(getResourcesDifference(resourcesA, resourcesA)).toEqual(resultA);
```

Comment:
can we just type both the objects and `toEqual<T>` (I believe it's `ResourcesWithUnits`, right ?) ?

Reply:
Done 👍